vispy/scene/cameras/cameras.py

```diff
@@ -1128,6 +1019,208 @@ def _update_projection_transform(self, fx, fy):
+        self._roll = roll
+        self.view_changed()
+
+    def orbit(self, azim, elev):
+        """ Orbits the camera around the center position.
+
+        Parameters
+        ----------
+        azim : float
+            Angle in degrees to rotate horizontally around the center point.
+        elev : float
+            Angle in degrees to rotate vertically around the center point.
+        """
+        self.azimuth += azim
+        self.elevation = np.clip(self.elevation + elev, -90, 90)
+        self.view_changed()
+
+    def _update_rotation(self, p1, p2):
+        """Update rotation parmeters based on mouse movement"""
+        if self._event_value is None:
+            self._event_value = self.azimuth, self.elevation
+        self.azimuth = self._event_value[0] - (p2 - p1)[0] * 0.5
+        self.elevation = self._event_value[1] + (p2 - p1)[1] * 0.5
+
+    def _rotate_tr(self, tr):
+        """Rotate the transformation matrix based on camera parameters"""
+        up, forward, right = self._get_dim_vectors()
+        tr.rotate(self.elevation, -right)
+        tr.rotate(self.azimuth, up)
+
+
+class ArcballCamera(Base3DRotationCamera):
+    """ 3D camera class that orbits around a center point while
+    maintaining a view on a center point.
+
+    For this camera, the ``scale_factor`` indicates the zoom level, and
+    the ``center`` indicates the position to put at the center of the
+    view.
+
+    Parameters
+    ----------
+    fov : float
+        Field of view. Zero (default) means orthographic projection.
+    distance : float | None
+        The distance of the camera from the rotation point (only makes sense
+        if fov > 0). If None (default) the distance is determined from the
+        scale_factor and fov.
+    See BaseCamera for more.
+
+    Interaction
+    -----------
+    * LMB: orbits the view around its center point.
+    * RMB or scroll: change scale_factor (i.e. zoom level)
+    * SHIFT + LMB: translate the center point
+    * SHIFT + RMB: change FOV
+    """
+
+    _state_props = Base3DRotationCamera._state_props
+
+    def __init__(self, fov=0.0, distance=None, **kwargs):
+        super(ArcballCamera, self).__init__(fov=fov, **kwargs)
+
+        # Set camera attributes
+        self._quaternion = Quaternion()
+        self.distance = distance  # None means auto-distance
+
+    def _update_rotation(self, p1, p2):
+        """Update rotation parmeters based on mouse movement"""
+        if self._event_value is None:
+            self._event_value = p2
+        wh = self._viewbox.size
+        self._quaternion = (self._quaternion *
+                            Quaternion(*_arcball(self._event_value, wh)) *
+                            Quaternion(*_arcball(p2, wh)))
+        self._event_value = p2
+        self.view_changed()
+
+    def _rotate_tr(self, tr):
+        """Rotate the transformation matrix based on camera parameters"""
+        tr.matrix = np.dot(self._quaternion.get_matrix().T, tr.matrix)
```

Comment:
I've applied some fixes to the Quaternion class in my DefaultTransform branch. I think one of them involved `get_matrix()`. I should either backport these fixes into master, or I'll need to fix this camera in that PR.